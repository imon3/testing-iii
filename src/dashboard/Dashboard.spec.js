// Test away
import React from 'react';
import { render, fireEvent } from 'react-testing-library';
import renderer from 'react-test-renderer';
import 'react-testing-library/cleanup-after-each';
import 'jest-dom/extend-expect';

import Dashboard from './Dashboard';

describe('<Dashboard />', () => {
    it('should show the controls', () => {
        const tree = renderer.create(<Dashboard />)

        expect(tree.toJSON()).toMatchSnapshot()
    });
});

describe('<Dashboard', () => {
    it('should show button text changes when clicked ', () => {
        const { getByText } = render(<Dashboard />);

        const closeButton = getByText(/close gate/i);
        fireEvent.click(closeButton);

        expect(getByText(/open gate/i)).toBeDefined();
    });
});