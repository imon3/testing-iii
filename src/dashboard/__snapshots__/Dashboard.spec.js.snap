// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> should show the controls 1`] = `
Array [
  <div
    className="display panel"
  >
    <div
      className="led green-led"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
  >
    <button
      className="toggle-btn"
      data-testid="lockButton"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      data-testid="closeButton"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
